src/rest-server/src/models/job.js

```diff
@@ -403,6 +403,14 @@ class Job {
     for (let i = 0; i < data.taskRoles.length; i ++) {
       tasksNumber += data.taskRoles[i].taskNumber;
     }
+    let jobEnvs = '';
+    if (data.jobEnvs) {
+        for (let key in data.jobEnvs) {
+            if (data.jobEnvs.hasOwnProperty(key)) {
+                jobEnvs = jobEnvs.concat(key, '=', data.jobEnvs[key], '\n');
```

Comment:
No encode needed?

Reply:
There are indeed injection risks, javascript injection and bash injection.
I will take more investigation.

Reply:
It's some kind of complicated.
We would pass the env.list to docker command, which would quote. 

So if we quote it with single/double quote, the value would corrupt, as below:
https://github.com/Microsoft/pai/blob/4491119258197ccd6700f295b3ff518ed4a83c81/src/rest-server/src/templates/yarnContainerScript.mustache#L277


```
hayua@stcvl-131:~$ cat env.list
double_quote="with double quote"
single_quote='with single quote'
hayua@stcvl-131:~$ sudo docker run -it --env-file env.list ubuntu env |grep quote
double_quote="with double quote"
single_quote='with single quote'
```

Reply:
Hi @Gerhut,
Now the "env list' was passing to 'docker run', which would take care of 'encoding and escaping'.
We should pass it as it is.